### PR TITLE
Rename schema to database

### PR DESCRIPTION
**Pull request by lanchengx**
Changes proposed in this pull request:
- Rename schema to database

### CODE REVIEW

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/ShowRulesStatementAssert.java [hunk 1]
```diff
@@ -136,8 +136,8 @@ public static void assertIs(final SQLCaseAssertContext assertContext, final Show
             ShowUnusedShardingAlgorithmsStatementAssert.assertIs(assertContext, (ShowUnusedShardingAlgorithmsStatement) actual, (ShowUnusedShardingAlgorithmsStatementTestCase) expected);
         } else if (actual instanceof ShowUnusedShardingKeyGeneratorsStatement) {
             ShowUnusedShardingKeyGeneratorsStatementAssert.assertIs(assertContext, (ShowUnusedShardingKeyGeneratorsStatement) actual, (ShowUnusedShardingKeyGeneratorsStatementTestCase) expected);
-        } else if (actual instanceof CountSchemaRulesStatement) {
-            CountSchemaRulesStatementAssert.assertIs(assertContext, (CountSchemaRulesStatement) actual, (CountSchemaRulesStatementTestCase) expected);
+        } else if (actual instanceof CountDatabaseRulesStatement) {
+            CountSchemaRulesStatementAssert.assertIs(assertContext, (CountDatabaseRulesStatement) actual, (CountSchemaRulesStatementTestCase) expected);
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/ShowRulesStatementAssert.java:140]: CountDatabaseRulesStatementAssert

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowAlgorithmsStatementAssert.java [hunk 2]
```diff
@@ -42,10 +42,10 @@ public final class ShowShadowAlgorithmsStatementAssert {
      */
     public static void assertIs(final SQLCaseAssertContext assertContext, final ShowShadowAlgorithmsStatement actual, final ShowShadowAlgorithmsStatementTestCase expected) {
         if (null != expected.getSchema()) {
-            assertTrue(assertContext.getText("Actual schema should exist."), actual.getSchema().isPresent());
-            SchemaAssert.assertIs(assertContext, actual.getSchema().get(), expected.getSchema());
+            assertTrue(assertContext.getText("Actual schema should exist."), actual.getDatabase().isPresent());
+            DatabaseAssert.assertIs(assertContext, actual.getDatabase().get(), expected.getSchema());
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowAlgorithmsStatementAssert.java:46]: expected.getDatabase()

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowTableRulesStatementAssert.java [hunk 3]
```diff
@@ -42,10 +42,10 @@ public final class ShowShadowTableRulesStatementAssert {
      */
     public static void assertIs(final SQLCaseAssertContext assertContext, final ShowShadowTableRulesStatement actual, final ShowShadowTableRulesStatementTestCase expected) {
         if (null != expected.getSchema()) {
-            assertTrue(assertContext.getText("Actual schema should exist."), actual.getSchema().isPresent());
-            SchemaAssert.assertIs(assertContext, actual.getSchema().get(), expected.getSchema());
+            assertTrue(assertContext.getText("Actual schema should exist."), actual.getDatabase().isPresent());
+            DatabaseAssert.assertIs(assertContext, actual.getDatabase().get(), expected.getSchema());
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowTableRulesStatementAssert.java:46]: expected.getDatabase())

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowTableRulesStatementAssert.java [hunk 4]
```diff
@@ -42,10 +42,10 @@ public final class ShowShadowTableRulesStatementAssert {
...
         } else {
-            assertFalse(assertContext.getText("Actual schema should not exist."), actual.getSchema().isPresent());
+            assertFalse(assertContext.getText("Actual schema should not exist."), actual.getDatabase().isPresent());
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShadowTableRulesStatementAssert.java:48]: Actual database

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShardingAlgorithmsStatementAssert.java [hunk 5]
```diff
@@ -42,10 +42,10 @@ public final class ShowShardingAlgorithmsStatementAssert {
      */
     public static void assertIs(final SQLCaseAssertContext assertContext, final ShowShardingAlgorithmsStatement actual, final ShowShardingAlgorithmsStatementTestCase expected) {
         if (null != expected.getSchema()) {
-            assertTrue(assertContext.getText("Actual schema should exist."), actual.getSchema().isPresent());
-            SchemaAssert.assertIs(assertContext, actual.getSchema().get(), expected.getSchema());
+            assertTrue(assertContext.getText("Actual schema should exist."), actual.getDatabase().isPresent());
+            DatabaseAssert.assertIs(assertContext, actual.getDatabase().get(), expected.getSchema());
         } else {
-            assertFalse(assertContext.getText("Actual schema should not exist."), actual.getSchema().isPresent());
+            assertFalse(assertContext.getText("Actual schema should not exist."), actual.getDatabase().isPresent());
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowShardingAlgorithmsStatementAssert.java:48]: Actual database

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowUnusedShardingKeyGeneratorsStatementAssert.java [hunk 6]
```diff
@@ -21,6 +21,7 @@
 import lombok.NoArgsConstructor;
 import org.apache.shardingsphere.sharding.distsql.parser.statement.ShowUnusedShardingKeyGeneratorsStatement;
 import org.apache.shardingsphere.test.sql.parser.parameterized.asserts.SQLCaseAssertContext;
+import org.apache.shardingsphere.test.sql.parser.parameterized.asserts.segment.schema.DatabaseAssert;
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/ShowUnusedShardingKeyGeneratorsStatementAssert.java:24]: segment.schema.DatabaseAssert - > segment.database.DatabaseAssert;

File: shardingsphere-distsql/shardingsphere-distsql-statement/src/main/java/org/apache/shardingsphere/distsql/parser/statement/rql/show/CountDatabaseRulesStatement.java [hunk 7]
```diff
@@ -19,15 +19,15 @@
 
 import lombok.Getter;
 import org.apache.shardingsphere.distsql.parser.subject.DistSQLSubjectTypeEnum;
-import org.apache.shardingsphere.sql.parser.sql.common.segment.generic.SchemaSegment;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.generic.DatabaseSegment;
 
 /**
  * Count schema rules statement.
```
Comment (TeslaCN) [inline shardingsphere-distsql/shardingsphere-distsql-statement/src/main/java/org/apache/shardingsphere/distsql/parser/statement/rql/show/CountDatabaseRulesStatement.java:25]: Missed here.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/CountDatabaseRulesStatementAssert.java [hunk 8]
```diff
@@ -31,21 +31,21 @@
  * Count schema rules statement assert.
```
Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/CountDatabaseRulesStatementAssert.java:31]: Missed comment.

Comment (RaigorJiang) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/distsql/rql/impl/rule/CountDatabaseRulesStatementAssert.java:31]: Missed comment.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**